e2etests/tests/unnumbered.go

```diff
@@ -0,0 +1,285 @@
+		cr := frrk8sv1beta1.FRRConfiguration{
+			ObjectMeta: metav1.ObjectMeta{
+				Name:      "for-" + externalP2PContainer.Name,
+				Namespace: k8s.FRRK8sNamespace,
+			},
+			Spec: frrk8sv1beta1.FRRConfigurationSpec{
+				BGP: frrk8sv1beta1.BGPConfig{
+					BFDProfiles: []frrk8sv1beta1.BFDProfile{{Name: "simple"}},
+					Routers: []frrk8sv1beta1.Router{
+						{
+							ASN: infra.FRRK8sASN,
+							Neighbors: []frrk8sv1beta1.Neighbor{
+								{
+									ASN:        remoteASN,
+									Interface:  iface,
+									BFDProfile: "simple",
+									ToReceive: frrk8sv1beta1.Receive{
+										Allowed: frrk8sv1beta1.AllowedInPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+									ToAdvertise: frrk8sv1beta1.Advertise{
+										Allowed: frrk8sv1beta1.AllowedOutPrefixes{
+											Mode: frrk8sv1beta1.AllowAll,
+										},
+									},
+								},
+							},
+							Prefixes: prefixSend,
+						},
+					},
+				},
+				NodeSelector: metav1.LabelSelector{
+					MatchLabels: map[string]string{
+						"kubernetes.io/hostname": nodeWithP2PConnection.GetLabels()["kubernetes.io/hostname"],
+					},
+				},
+			},
+		}
+		err = updater.Update([]corev1.Secret{}, cr)
+		Expect(err).NotTo(HaveOccurred())
+		ginkgo.DeferCleanup(func() {
+			err := updater.DeleteFRRConfiguration(cr)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		nodeP2PContainer := executor.ForContainer(nodeWithP2PConnection.Name)
+		nodeLLA, err := netdev.LinkLocalAddressForDevice(nodeP2PContainer, iface)
+		Expect(err).NotTo(HaveOccurred())
+		peerLLA, err := netdev.LinkLocalAddressForDevice(externalP2PContainer, iface)
+		Expect(err).NotTo(HaveOccurred())
+
+		ginkgo.By("validating the node and p2p container peered")
+		validateUnnumberedBGPPeering(externalP2PContainer, nodeLLA)
+
+		ginkgo.By(fmt.Sprintf("validating the p2p peer %s received the routes from node", externalP2PContainer.Name))
+		validatePrefixViaFor(externalP2PContainer, iface, nodeLLA, prefixSend)
+
+		ginkgo.By(fmt.Sprintf("validating the node received %s received the routes from p2p peer", nodeWithP2PConnection.Name))
+		validatePrefixViaFor(nodeP2PContainer, iface, peerLLA, rc.ToAdvertiseV4, rc.ToAdvertiseV6)
+	},
+		ginkgo.Entry("iBGP", frrconfig.RouterConfigUnnumbered{
+			ASNLocal:      infra.FRRK8sASN,
+			ASNRemote:     infra.FRRK8sASN,
+			Hostname:      "tor1",
+			Interface:     "net1",
+			RouterID:      "1.1.1.1",
+			ToAdvertiseV4: []string{"1.1.1.0/24"},
+			ToAdvertiseV6: []string{"1111::/64"},
+		}),
+		ginkgo.Entry("eBGP", frrconfig.RouterConfigUnnumbered{
+			ASNLocal:      4200000000,
+			ASNRemote:     infra.FRRK8sASN,
+			Hostname:      "tor2",
+			Interface:     "net2",
+			RouterID:      "2.2.2.2",
+			ToAdvertiseV4: []string{"2.2.2.0/24"},
+			ToAdvertiseV6: []string{"2222::/64"},
+		}),
+	)
+
+	// This spec is added to verify that the point-to-point setup works
```

Comment:
Not sure I understood. What's the difference with the other test (apart from using the raw config?)

Reply:
This test is actually checking  that the create container, the p2p wire function , and the unnumber peer config  works, without depending on actual code.

E.g. if that test is failing we should be looking whether test code change, external peer config, docker version, frr version introduced something. 

Do you prefer that I remove it or update the comment?

Reply:
Let's remove it. I understand what you mean, but the point of having a CI is to run it incrementally every change and have vision of what's breaking and when. So ideally, if changing a test or frr or docker version is breaking, we'll find out when we apply that change.

Under this logic we should duplicate all our tests to use raw config too.